### DISCUSSION TITLE
max-w-screen-xl not working when setting max in breakpoints ?

### DISCUSSION THREAD

**Discussion by DavidSvi** [tailwindlabs/tailwindcss#9247] [Help]
HI there, I have followed the tutorial on this page to config tailwind to go for desktop-first approach

https://tailwindcss.com/docs/screens

my config file is:

```
module.exports = {
	darkMode: 'class',
	theme: {
		extend: {},
		
	        screens: {					
                   'xl' : {'max':'250px'}
    	        },
	},
	
	
	plugins: [],
	corePlugins: {
	},
	  
 
}
```

I am setting max for multiple breakpoints, when I try to add `max-w-screen-xl` in the CSS 

`.class {
@apply mx-auto max-w-screen-xl
}`

I get this message: The `max-w-screen-xl` class does not exist. If `max-w-screen-xl` is a custom class, make sure it is defined within a `@layer` directive.

However, if I removed max from the config file I get no error and it works well.

How can I resolve this issue ?

Thank you

Tailwind Play: https://play.tailwindcss.com/eYgfJfxmUV?file=config

**Comment by wongjn** [comment]
The `max-w-screen-*` are only generated for `screens` values that are strings. So if you want `max-w-screen-xl`, you will need to rewrite your `xl` screen so it is a string value (and thus a `min-width` media query) or create a different screen key that works for your needs.

**Reply by ornakash** [reply]
I tried that but it doesn't work unfortunately.. do you have anther solution maybe?
![image](https://github.com/tailwindlabs/tailwindcss/assets/45389557/054a0e92-754b-49a8-b9de-9a2d1ccdbab2)

**Reply by wongjn** [reply]
What doesn't work for you? Which class are you trying to use?

**Reply by ornakash** [reply]
@wongjn thanks for the response!

I have this "tall" screen, once I put it, 
I can't use max-md, max-lg, etc. They seem to not work anymore

for instance:
className="max-md:pt-8 pt-3" used to work but when I add the "tall" screen, it doesn't work anymore


![image](https://github.com/tailwindlabs/tailwindcss/assets/45389557/0a8d596c-538d-4103-86f9-1df3d55f82e1)

**Reply by wongjn** [reply]
When you have `screens` values that are no longer string-only values, the `max-*` and `min-*` variants no longer exist, see #9558.

**Comment by ornakash** [comment]
Same issue here
